server/utils/referrals/caseListUtils.test.ts

```diff
@@ -159,126 +159,259 @@ describe('CaseListUtils', () => {
-            html: '<a class="govuk-link" href="/assess/referrals/referral-123/personal-details">Del Hatton</a><br>ABC1234',
-          },
-          {
-            attributes: { 'data-sort-value': undefined },
-            text: 'N/A',
-          },
-          {
-            attributes: {
-              'data-sort-value': '2022-01-01T00:00:00.000Z',
-            },
-            text: '1 January 2022',
-          },
-          {
-            attributes: {
-              'data-sort-value': undefined,
-            },
-            text: 'N/A',
-          },
-          {
-            attributes: { 'data-sort-value': '2023-01-01T00:00:00.000Z' },
-            text: '1 January 2023',
-          },
-          {
-            attributes: { 'data-sort-value': '2022-01-01T00:00:00.000Z' },
-            text: '1 January 2022',
-          },
-          {
-            attributes: { 'data-sort-value': '2024-01-01T00:00:00.000Z' },
-            text: '1 January 2024',
-          },
-          {
-            attributes: { 'data-sort-value': 'Whatton (HMP)' },
-            text: 'Whatton (HMP)',
-          },
-          { text: 'Test Course 1' },
-          {
-            text: 'General offence',
-          },
-          {
-            attributes: { 'data-sort-value': 'referral_started' },
-            html: CaseListUtils.statusTagHtml('referral_started'),
-          },
-        ],
-        [
-          {
-            attributes: { 'data-sort-value': '' },
-            html: '<a class="govuk-link" href="/assess/referrals/referral-456/personal-details">DEF1234</a>',
-          },
-          {
-            attributes: { 'data-sort-value': '2021-01-01T00:00:00.000Z' },
-            text: '1 January 2021',
-          },
-          {
-            attributes: { 'data-sort-value': undefined },
-            text: 'N/A',
-          },
-          {
-            attributes: { 'data-sort-value': undefined },
-            text: 'N/A',
-          },
-          {
-            attributes: { 'data-sort-value': undefined },
-            text: 'N/A',
-          },
-          {
-            attributes: { 'data-sort-value': undefined },
-            text: 'N/A',
-          },
-          {
-            attributes: { 'data-sort-value': undefined },
-            text: 'N/A',
-          },
-          {
-            attributes: { 'data-sort-value': undefined },
-            text: 'N/A',
-          },
-          { text: 'Test Course 2' },
-          {
-            text: 'General offence, Extremism offence',
-          },
-          {
-            attributes: { 'data-sort-value': 'referral_submitted' },
-            html: CaseListUtils.statusTagHtml('referral_submitted'),
-          },
-        ],
-      ])
+    describe('when passed the headings for Assess case lists', () => {
+      it('formats referral summary information in the appropriate format for passing to a GOV.UK table Nunjucks macro, based on the provided headings', () => {
```

Comment:
Yeah we could do that - do you think it's okay just to have a single test with a subset of all possible columns passed in then?

Reply:
I think so, yeah

Reply:
I've revised this a little so that it checks all fields in one test (since we want to check the rest of the object looks right, not just the bits generated by `tableRowContent`), then checks that filtering works in another with just three columns passed in